starfish/imagestack/imagestack.py

```diff
@@ -290,6 +290,131 @@ def xarray(self) -> xr.DataArray:
+
+        Parameters
+        ----------
+        indexers : a dictionary of dim:index where index is the value
+        or range to index the dimension
+
+        """
+        new_coords = self._coordinates.copy()
+        # index by R, CH, V
+        key = (indexers[Indices.ROUND.value], indexers[Indices.CH.value], indexers[Indices.Z.value])
+        new_coords = self._index_keep_dimensions(new_coords, key)
+        # check if X or Y dimension indexed, if so rescale
+        if self._needs_coords_resacling(indexers):
+            self._rescale_physical_coordinates(indexers, new_coords)
+        return new_coords
+
+    def _rescale_physical_coordinates(self, indexers: dict, new_coords: xr.DataArray) -> None:
+        """Iterates through coordinates array and rescales x, y physical coordinate values
+         based on how the x and y dimensions were indexed
+
+        Parameters
+        ----------
+        indexers : a dictionary of dim:index where index is the value
+        or range to index the dimension
+
+        new_coords: the coordinates xarray to modify
+
+        """
+        for _round in range(new_coords.sizes[Indices.ROUND]):
+            for ch in range(new_coords.sizes[Indices.CH]):
+                for z in range(new_coords.sizes[Indices.Z]):
+                    indices = {
+                        Indices.ROUND: _round,
+                        Indices.CH: ch,
+                        Indices.Z: z
+                    }
+                    xmin, xmax = self._rescale_physical_coordinate(
+                        Coordinates.X,
+                        indices,
+                        indexers[Indices.X.value])
+                    ymin, ymax = self._rescale_physical_coordinate(
+                        Coordinates.Y,
+                        indices,
+                        indexers[Indices.Y.value])
+
+                    new_coords[indices].loc[
+                        dict(physical_coordinate=[
+                            PhysicalCoordinateTypes.X_MIN,
+                            PhysicalCoordinateTypes.X_MAX,
+                            PhysicalCoordinateTypes.Y_MIN,
+                            PhysicalCoordinateTypes.Y_MAX]
+                        )] = [xmin, xmax, ymin, ymax]
+
+    def _rescale_physical_coordinate(self, coord: Coordinates, indices: Mapping[Indices, int],
+                                     key) -> Tuple:
+        """Calculates rescaled coordinates
+
+        Parameters
+        ----------
+        coord: The coordinate to rescale.
+
+        indices: The (Round, Ch, Z) indices that identify which tile's coordinates we're using
+
+        key: a value or range to index on
+
+        """
+        coord_min, coord_max = self.coordinates(indices, coord)
+        dimension_size = self.xarray.sizes[coord.value[0]]
+        physical_pixel_size = (coord_max - coord_min) / dimension_size
+        new_min, new_max = coord_min, coord_max
+        min_pixel_index = key if type(key) is int else key.start
```

Comment:
This is disambiguating between an int and a `slice`?

Reply:
Yea when the indexing comes in as a slice like [:5] it's turned into a python Slice object which is basically just a tuple but the object has start and stop attributes that are the start and stop of the slicing operation.

Reply:
Cool, store as a line comment please. :-)